src/OpenTelemetry.Sampler.AWS/SamplingRuleApplier.cs

```diff
@@ -107,12 +162,91 @@ public bool Matches(SamplingParameters samplingParameters, Resource resource)
                Matcher.WildcardMatch(GetArn(in samplingParameters, resource), this.Rule.ResourceArn);
     }
 
-    [SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "method work in progress")]
     public SamplingResult ShouldSample(in SamplingParameters samplingParameters)
     {
-        // for now return drop sampling result.
-        // TODO: use reservoir and fixed rate sampler
-        return new SamplingResult(false);
+        Interlocked.Increment(ref this.Statistics.RequestCount);
+        bool reservoirExpired = this.Clock.Now() >= this.ReservoirEndTime;
```

Comment:
I wonder if we should log when we make a sampling decision and the reservoir is expired?

Reply:
Thanks is a good point.